packages/widgetbook/lib/src/fields/duration_field.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+
+import 'field.dart';
+import 'field_codec.dart';
+import 'field_type.dart';
+
+class DurationField extends Field<Duration> {
+  DurationField({
+    required super.name,
+    required Duration super.initialValue,
+    super.onChanged,
+  }) : super(
+          type: FieldType.doubleInput,
+          codec: durationCodec,
```

Comment:
Although I understand the reason behind extracting this to its own variable to make it more testable. I would be more leaning towards adding it to the constructor directly, just like other fields, to maintain the same code style everywhere.

_(Note that I have used the same param names, branching style, etc. from other fields)_

```suggestion
          codec: FieldCodec(
            toParam: (value) => value.inMilliseconds.toString(),
            toValue: (param) {
              return param == null
                  ? null
                  : Duration(
                      milliseconds: int.tryParse(param) ?? 0,
                    );
            },
          ),
```